### PR TITLE
#1738 strip HTML, newlines, and tabs from Player input text

### PR DESCRIPTION
**Pull request by trhr**
#### Brief overview of PR changes/additions
Patched security issue that allowed players to spoof by adding newlines. Will not affect developers, and easily overloaded by server devs who desire different functionality.

#### Motivation for adding to Evennia
1.0 gogogo

#### Other info (issues closed, discussion etc)
I wonder if this addresses MXP security issue? Surely you'd need to include HTML in your MXP link to actually execute arbitrary code from it.

### CODE REVIEW

File: evennia/server/inputfuncs.py [hunk 1]
```diff
@@ -41,6 +42,28 @@
...
+        txt (str) : A cleansed input string
+    """
+    if session.account:
+        if session.account.permissions.get("Developer"):
+            return txt
+
+    # Strip unsafe characters from ingest
+    txt = strip_tags(txt)  # Any HTML codes
+    txt = re.sub(r'|\/', '', txt)  # New lines
+    txt = re.sub(r'|-', '', txt)  # Tabs
```
Comment (Griatch) [inline evennia/server/inputfuncs.py:65]: This is highly specific to Evennia's default ANSI formatter and people can replace it at will (there is even a contrib to do it). Hard-coding these sequences here is not useful. 

Besides, these formatting instructions are expected by users - a user entering a description string in any more RP-heavy game will expect the ability to at least be able to divvy up their text into paragraphs with line break. If we remove them here, what's the point of having these formatting operations at all (if you produce this in code you can just use \n \t directly)?

File: evennia/server/inputfuncs.py [hunk 2]
```diff
@@ -41,6 +42,28 @@
 def _NA(o):
     return "N/A"
 
+def _strip_unsafe(session, txt):
```
Comment (Griatch) [inline evennia/server/inputfuncs.py:45]: If this is intended to be overridden, it's best to leave it named without the leading `_`. We have not been very consistent with this everywhere, but overall, "_private" methods tends to be something that is not really a part of the public API, and which can change more willy-nilly (this is the use in e.g. Twisted and Django too).

**Comment by Griatch** [comment]
@trhr What's your view of my comments above? What is the plan forward for this PR?

**Comment by trhr** [comment]
To be addressed at the channel level.

**Pull request closed without merging**